src/entity.unit.ts

```diff
@@ -344,4 +426,90 @@ describe("Entity", () => {
 			expect(movie.toString("[Director]")).toBe("Ridley Scott $100,000.00");
 		});
 	});
+
+	describe("markPersisted()", () => {
+		it("assigning identifier property of entity calls markPersisted", () => {
+			const movie = new Types.Movie(Alien);
+			const markPersisted = jest.spyOn(movie, "markPersisted");
+			movie.Id = "1";
+			expect(markPersisted).toBeCalledTimes(1);
+		});
+
+		it("does not affect non-identifying entity", () => {
+			const budget = new Types.Budget();	// Budget type has no identifier property
+			budget.markPersisted();
+			expect(budget.meta.isNew).toBeTruthy();
+		});
+
+		describe("identifying entity", () => {
+			let movie: Entity;
+
+			beforeEach(() => {
+				movie = new Types.Movie(Alien);	// Movie type has an identifier property
+				movie.update({
+					Budget: {
+						LineItems: [
+							{ Label: "Item 1", Cost: 100 }
+						]
+					}
+				});
+			});
+
+			describe("with no id", () => {
+				it("does nothing", () => {
+					movie.markPersisted();
+					expect(movie.meta.isNew).toBeTruthy();
+				});
+
+				it("does not affect nested entities", () => {
+					movie.markPersisted();
+
+					expect(movie.Budget.meta.isNew).toBeTruthy();
+					for (const lineItem of movie.Budget.LineItems)
+						expect(lineItem.meta.isNew).toBeTruthy();
+				});
+			});
+
+			describe("with id", () => {
+				beforeEach(() => {
+					movie.Id = "1";
+				});
+
+				it("marks nested, non-identifying entities as not new", () => {
+					movie.markPersisted();
+
+					expect(movie.Budget.meta.isNew).toBeFalsy();
+					for (const lineItem of movie.Budget.LineItems)
+						expect(lineItem.meta.isNew).toBeFalsy();
+				});
+
+				it("handles circular references", () => {
+					movie.update({ Credits: { Movie: movie } });
+
+					movie.markPersisted();
+
+					expect(movie.Credits.meta.isNew).toBeFalsy();
+				});
+
+				it("does not affect nested, sovereign entities", () => {
+					movie.markPersisted();
+					expect(movie.Director.meta.isNew).toBeTruthy();
+
+					movie.Director.update({
+						Id: "1",
+						// TODO: should Address be new after this update operation, even though Director's Id is getting assigned?
+						Address: { City: "Orlando", State: "Florida" }
+					});
```

Comment:
See the TODO comment. Currently, in this example, `Address.meta.isNew === true` after this update occurs, because the id assignment technically occurs before the Address entity is created and attached to the Director. I don't know if that is a bad thing, but could definitely see it going either way.

Reply:
Hmm, good question. I think from a user perspective I would expect it to no longer be new, since it appears as if setting the `Id` and setting the `Address` happen at the same time, or as a part of the same operation. Also, I could see this being the result of a server-side call which includes the objects persisted id as well as some additional data. Not sure how big of a deal it is though, especially depending on how much effort it would take to change. Also, is it true that you could always call `markPersisted` after setting the id?

Reply:
I've modified the `update()` method to include a call to `markPersisted` after the update occurs so that owned entities created during an update that includes an id assignment are _not considered new_. Added a couple tests for this too.